src/allocator.f90

```diff
@@ -170,12 +180,28 @@ function get_block(self, dir) result(handle)
     select case (dir)
     case (DIR_X)
       dims = self%xdims_padded
+      handle%n = self%cdims(1)
+      handle%SZ = self%xdims_padded(1)
+      handle%n_padded = self%xdims_padded(2)
+      handle%n_groups = self%xdims_padded(3)
     case (DIR_Y)
       dims = self%ydims_padded
+      handle%n = self%cdims(2)
+      handle%SZ = self%ydims_padded(1)
+      handle%n_padded = self%ydims_padded(2)
+      handle%n_groups = self%ydims_padded(3)
     case (DIR_Z)
       dims = self%zdims_padded
+      handle%n = self%cdims(3)
+      handle%SZ = self%zdims_padded(1)
+      handle%n_padded = self%zdims_padded(2)
+      handle%n_groups = self%zdims_padded(3)
     case (DIR_C)
       dims = self%cdims_padded
+      handle%n = -1
+      handle%SZ = -1
+      handle%n_padded = -1
+      handle%n_groups = -1 
```

Comment:
I think this also suggests that we shouldn't store these variables in `field_t`. `field_t` is effectively a scratch space and sometimes these variables don't have a meaning. These assignments in particular can cause annoying bugs.